app/src/main/java/nodomain/freeyourgadget/gadgetbridge/service/devices/miband2/operations/FetchActivityOperation.java

```diff
@@ -207,6 +210,14 @@ private GregorianCalendar saveSamples() {
 
                     timestamp.add(Calendar.MINUTE, 1);
                 }
+
+                // export csv on fetching
+                File myPath = null;
+                myPath = FileUtils.getExternalFilesDir();
+                File myFile = new File(myPath, startTimestamp.getTimeInMillis() / 1000 + ".csv");
```

Comment:
Maybe add a `gadgetbridgedb_` prefix (or something like that, maybe shorter)?

But really, this does not belong here. It should be implemented in one place for all devices, not in the Mi Band 2 FetchActivityOperation. And I guess we should also have an option to turn this on and off.

Reply:
Right, does data from other devices differs ? Where would you place on/off feature ?

Reply:
Data from other devices may differ, yes. So either we use normalized values (= the same data ranges for all devices) or device specific raw values, or even both.

But it would be good to have this for all devices, not just one.

Since some devices provide the data in a different way, where we have no access to all samples at once, we might call a method like `samplesAdded(device, start, end)` which would then check if CSV export is enabled, then fetch the samples from the given date range from the db (probably cached in the session already) and perform the export.

Alternatively we could integrate with the new DB export functionality and perform the CSV export repeatedly. We would just need to remember the start and end dates so that we only export the incremental changes.